### ISSUE TITLE
Simple query document string exports

### ISSUE THREAD

**Issue by LeviSchuck** [dotansimha/graphql-code-generator#1978]
It would be useful to have parsable query to send to clients or to execute locally while also in a typed environment.

In my case I'm using graphql directly on an executable schema and I rely on both an input query, types on the input data, and types on the output.

Configurations that might be useful to avoid collisions would be
* Postfix
* Prefix

Here's a proof of concept that I am using in my own project for the time being.

```js
const {print} = require('graphql/language/printer');

module.exports = {
  plugin: (unused, documents) => {
    const encounteredQueries = {};
    for (let i = 0; i < documents.length; i++) {
      const defs = documents[i].content.definitions;
      for (let j = 0; j < defs.length; j++) {
        if (defs[j].operation !== 'query') {
          console.log("Skipping non-query", defs[j]);
          continue;
        }
        const name = defs[j].name.value;
        if (!name) {
          console.warn("Skipping unnamed", defs[j]);
        }
        if (encounteredQueries[name]) {
          console.warn("Skipping duplicate query by ", name, documents[i])
          continue;
        }
        try {
          encounteredQueries[name] = defs[j];
        } catch (e) {
          console.error(e);
        }
        
      }
    }
    return `
// This is at auto generated file, it contains exported queries

${Object.values(encounteredQueries).map((query) => {
  return `export const ${query.name.value}GraphQuery = \`
${print(query)}
\`;`
}).join('\n\n')}
    `;
  }
}
```

**Comment by dotansimha** [comment]
Hi @LeviSchuck, If you are in a typed environment (such as TypeScript), you can use `typescript-operations` and it will create the exact thing for you (along with typescript interfaces for your queries response and variables).

See an example here: https://github.com/dotansimha/graphql-code-generator/blob/master/dev-test/githunt/types.reactApollo.tsx#L314 

If you wish to have it as a standalone plugin, we always welcome PRs :) (and btw, no need to check for unnamed or duplicated because the codegen does that for you before running the plugins)

**Comment by dotansimha** [comment]
Implemented here: https://github.com/dotansimha/graphql-code-generator/pull/2098

**Comment by morenore** [comment]
Ok

**Comment by dotansimha** [comment]
Available in 1.4.0 🎉
